src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/DefaultTagHelperCompletionService.cs

```diff
@@ -133,12 +143,20 @@ public override IReadOnlyList<CompletionItem> GetCompletionsAt(SourceSpan locati
                     filterText = filterText.Substring(0, filterText.Length - 3);
                 }
 
-                var insertTextFormat = InsertTextFormat.Snippet;
-                if (!TryResolveAttributeInsertionSnippet(filterText, completion.Value, indexerCompletion, out var insertText))
+                string insertText;
+                InsertTextFormat insertTextFormat;
+
+                var snippetSupported = LanguageServer.ClientSettings?.Capabilities?.TextDocument?.Completion.Value?.CompletionItem?.SnippetSupport ?? false;
```

Comment:
Why not bake this piece into `TryResolveAttributeInsertionSnippet` and make it non-static?